pkg/k8s/scanner/scanner.go

```diff
@@ -143,3 +176,140 @@ func (s *Scanner) filter(ctx context.Context, r types.Report, artifact *artifact
 	}
 	return report.CreateResource(artifact, r, nil), nil
 }
+
+func clusterInfoToReportResources(allArtifact []*artifacts.Artifact, clusterName string) (*core.Component, error) {
+	coreComponents := make([]*core.Component, 0)
+	for _, artifact := range allArtifact {
+		switch artifact.Kind {
+		case pod:
+			var comp bom.Component
+			err := ms.Decode(artifact.RawResource, &comp)
+			if err != nil {
+				return nil, err
+			}
+			packages := make(ftypes.Packages, 0)
+			repoDigest := make([]string, 0)
+			for _, c := range comp.Containers {
+				name := fmt.Sprintf("%s/%s", c.Registry, c.Repository)
+				version := sanitizedVersion(c.Version)
+				packages = append(packages, ftypes.Package{
+					ID:      fmt.Sprintf("%s:%s", name, version),
+					Name:    name,
+					Version: version,
+					Digest:  digest.NewDigestFromString(digest.SHA256, strings.ReplaceAll(c.Digest, "sha256:", "")),
+				},
+				)
+			}
+			podReport := types.Report{
+				ArtifactName: comp.Name,
+				ArtifactType: ftypes.KubernetesPod,
+				Metadata: types.Metadata{
+					RepoDigests: repoDigest,
+				},
+				Results: types.Results{
+					{
+						Target:   "containers",
+						Type:     "oci",
+						Class:    types.ClassK8sComponents,
+						Packages: packages,
+					},
+				},
+			}
+			podComp, err := cyclonedx.MarshalReport(podReport)
```

Comment:
I don't think we need to call `MarshalReport` here. You can directly construct core.Component.

```
podComp := core.Component {
	Type: cdx.Container,
	Name: comp.Name,
        ...
	Properties: comp.Properties,
}
```